File: src/main/java/org/flighthub/Repository/IAgentRepository.java
package org.flighthub.Repository;

import org.flighthub.Domain.Agent;

import java.util.Optional;
import java.util.UUID;

public interface IAgentRepository extends Repository<UUID, Agent> {
    Optional<Agent> findOne(UUID id);

    Iterable<Agent> findAll();

    Agent save(Agent entity);

    Agent save(Agent entity, String password);

    Optional<Agent> findOneByUsername(String username);

    Optional<Agent> delete(UUID id);

    Optional<Agent> update(Agent entity);

    Agent login(String username, String password);
}

File: Server - C#/persistence/IAgentRepository.cs
using FlightHubC_.Domain;
using System;
using System.Collections.Generic;

namespace FlightHubC_.Repository
{
    public interface IAgentRepository : IRepository<Guid, Agent>
    {
        Agent FindOne(Guid id);
        Agent FindByUsername(string username);
        IEnumerable<Agent> FindAll();
        Agent Save(Agent entity);
        Agent Save(Agent entity, String password);
        Agent Delete(Guid id);
        Agent Update(Agent entity);
        Agent Login(string username, string password);

    }
}
